app/api/server/v1/voip/rooms.ts

```diff
@@ -0,0 +1,182 @@
+			agentId: Match.Maybe(String),
+		};
+		check(this.queryParams, defaultCheckParams);
+
+		const { token, rid, agentId } = this.queryParams;
+		const guest = await LivechatVisitors.getVisitorByToken(token, {});
+		if (!guest) {
+			throw new Meteor.Error('invalid-token');
+		}
+
+		let room;
+		if (!rid) {
+			room = await VoipRoom.findOneOpenByVisitorToken(token, { projection: API.v1.defaultFieldsToExclude });
+			if (room) {
+				return API.v1.success({ room, newRoom: false });
+			}
+			let agent;
+
+			let agentObj = null;
+			if (agentId) {
+				agentObj = await LivechatVoip.findAgent(agentId);
+			}
+			if (agentObj) {
+				const { username } = agentObj;
+				agent = { agentId, username };
+			}
+			const rid = Random.id();
+			const roomInfo = {
+				source: {
+					type: OmnichannelSourceType.API,
+				},
+			};
+			room = await LivechatVoip.getNewRoom(guest, agent, rid, roomInfo, { projection: API.v1.defaultFieldsToExclude });
+			return API.v1.success(room);
+		}
+		room = await VoipRoom.findOneOpenByRoomIdAndVisitorToken(rid, token, { projection: API.v1.defaultFieldsToExclude });
+		if (!room) {
+			throw new Meteor.Error('invalid-room');
+		}
+		return API.v1.success({ room, newRoom: false });
+	},
+});
+
+/**
+ * @openapi
+ *  /voip/server/api/v1/voip/room.close <AMOL Verify during code review>
+ *    post:
+ *      description: Closes an open room
+ * 		based on rid and token
+ *      security:
+ *		requestBody:
+ *      required: true
+ *      content:
+ *			application/json:
+ *          schema:
+ *          	type: object
+ *			  	properties:
+ *					rid:
+ *                 		type: string
+ *					token:
+ *						type: string
+ *      responses:
+ *        200:
+ *          description: rid of closed room and a comment for closing room
+ *          content:
+ *            application/json:
+ *              schema:
+ *                allOf:
+ *                  - $ref: '#/components/schemas/ApiSuccessV1'
+ *                  - type: object
+ *                    properties:
+ *                      rid:
+ *                        	type: string
+ *                      comment:
+ *                      	type: string
+ *        default:
+ *          description: Unexpected error
+ *          content:
+ *            application/json:
+ *              schema:
+ *                $ref: '#/components/schemas/ApiFailureV1'
+ */
+API.v1.addRoute('voip/room.close', {
+	async post() {
+		try {
+			check(this.bodyParams, {
+				rid: String,
+				token: String,
+			});
+			const { rid, token } = this.bodyParams;
+
+			const visitor = await LivechatVisitors.getVisitorByToken(token, {});
+			if (!visitor) {
+				throw new Meteor.Error('invalid-token');
+			}
+			const roomResult = await LivechatVoip.findRoom(token, rid);
+			if (!roomResult) {
+				throw new Meteor.Error('invalid-room');
+			}
+			const room: IVoipRoom = roomResult;
+			if (!room.open) {
+				throw new Meteor.Error('room-closed');
+			}
+			const language = 'en';
```

Comment:
Should this language be loaded from somewhere?

Reply:
I copied it from app/livechat/server/api/v1/room.js. Not sure why I deleted the reading of language from settings. Adding it now